app/assets/javascripts/discourse/app/components/topic-timeline/container.js

```diff
@@ -0,0 +1,331 @@
+    return this.lastReadTop > bottom ? bottom : this.lastReadTop;
+  }
+
+  get bottomAge() {
+    return relativeAge(
+      new Date(this.args.model.last_posted_at || this.args.model.created_at),
+      {
+        addAgo: true,
+        defaultFormat: timelineDate,
+      }
+    );
+  }
+
+  get startDate() {
+    return timelineDate(this.args.model.createdAt);
+  }
+
+  get nowDate() {
+    return this.bottomAge;
+  }
+
+  get lastReadHeight() {
+    return Math.round(this.lastReadPercentage * scrollareaHeight());
+  }
+
+  @bind
+  calculatePosition() {
+    this.timelineScrollareaStyle = htmlSafe(`height: ${scrollareaHeight()}px`);
+
+    const topic = this.args.model;
+    const postStream = topic.postStream;
+    this.total = postStream.filteredPostsCount;
+
+    this.scrollPosition =
+      this.clamp(Math.floor(this.total * this.percentage), 0, this.total) + 1;
+
+    this.current = this.clamp(this.scrollPosition, 1, this.total);
+    const daysAgo = postStream.closestDaysAgoFor(this.current);
+
+    let date;
+    if (daysAgo === undefined) {
+      const post = postStream.posts.findBy(
+        "id",
+        postStream.stream[this.current]
+      );
+
+      if (post) {
+        date = new Date(post.created_at);
+      }
+    } else if (daysAgo !== null) {
+      date = new Date();
+      date.setDate(date.getDate() - daysAgo || 0);
+    } else {
+      date = null;
+    }
+
+    this.date = date;
+
+    const lastReadId = topic.last_read_post_id;
+    const lastReadNumber = topic.last_read_post_number;
+
+    if (lastReadId && lastReadNumber) {
+      const idx = postStream.stream.indexOf(lastReadId) + 1;
+      this.lastRead = idx;
+      this.lastReadPercentage = this._percentFor(topic, idx);
+    }
+
+    if (this.position !== this.scrollPosition) {
+      this.position = this.scrollPosition;
+      this.updateScrollPosition(this.current);
+    }
+
+    this.before = this.scrollareaRemaining() * this.percentage;
+    this.after = scrollareaHeight() - this.before - SCROLLER_HEIGHT;
+
+    if (this.percentage === null) {
+      return;
+    }
+
+    if (this.hasBackPosition) {
+      this.lastReadTop = Math.round(
+        this.lastReadPercentage * scrollareaHeight()
+      );
+      this.showButton =
+        this.before + SCROLLER_HEIGHT - 5 < this.lastReadTop ||
+        this.before > this.lastReadTop + 25;
+    }
+
+    if (this.hasBackPosition) {
+      this.lastReadTop = Math.round(
+        this.lastReadPercentage * scrollareaHeight()
+      );
+    }
+  }
+
+  @bind
+  updateScrollPosition(scrollPosition) {
+    // only ran on mobile
+    if (!this.args.fullscreen) {
+      return;
+    }
+
+    const stream = this.args.model.postStream;
+
+    // a little debounce to avoid flashing
+    discourseLater(() => {
+      if (!this.position === scrollPosition) {
+        return;
+      }
+
+      // we have an off by one, stream is zero based,
+      stream.excerpt(scrollPosition - 1).then((info) => {
```

Comment:
In local testing, `scrollPosition` is `NaN` which causes a whole load of ajax errors in the console

Reply:
I am not able to replicate this on mobile or desktop... could you test again?

Reply:
See last commit 👍